api/controllers/documentsV2.js

```diff
@@ -0,0 +1,577 @@
+        }
+    }
+    catch (e)
+    {
+        defaultLog.error('### Error in {PUT}/Documents/{id}/Feature :', e);
+        return Actions.sendResponseV2(res, 500, { code: '500', message: 'Internal Server Error', self: 'Api/Documents/{id}/Feature' });        
+    }
+    finally
+    {
+        defaultLog.debug('<<< {PUT}/Documents/{id}/Feature');
+    }
+};
+
+// PUT (Secure, unfeatureDocument)
+exports.unfeatureDocument = async function (args, res, next)
+{
+    defaultLog.debug('>>> {PUT}/Documents/{id}/Unfeature');
+
+    try
+    {
+        if (args.swagger.params.hasOwnProperty('docId'))
+        {
+            let documentId = args.swagger.params.docId.value;
+            let document = await documentDAO.fetchDocument(documentId);
+
+            if (document)
+            {
+                document = await documentDAO.unfeatureDocument(document);
+
+                if(!document.isFeatured)
+                {
+                    document = documentDAO.documentHateoas(updatedDocument, constants.SECURE_ROLES);
+                    return Actions.sendResponseV2(res, 200, document);
+                }
+                else
+                {
+                    throw Error('Could not un-feature document');
+                }
+            }
+            else
+            {
+                throw Error('Document does not have a project reference');
+            }
+        }
+    }
+    catch (e)
+    {
+        defaultLog.error('### Error in {PUT}/Documents/{id}/Unfeature :', e);
+        return Actions.sendResponseV2(res, 500, { code: '500', message: 'Internal Server Error', self: 'Api/Documents/{id}/Unfeature' });        
+    }
+    finally
+    {
+        defaultLog.debug('<<< {PUT}/Documents/{id}/Unfeature');
+    }
+};
+
+// GET (Secure, downloadDocument)
+exports.downloadDocumentSecure = async function(args, res, next)
+{
+    defaultLog.debug('>>> {GET}/Documents/{id}/Download');
+
+    try
+    {
+        if (args.swagger.params.hasOwnProperty('docId'))
+        {
+            let userName = args.swagger.params.auth_payload.preferred_username;
+            let documentId = args.swagger.params.docId.value;
+            let document = await documentDAO.fetchDocument(documentId);
+            let fileNameOverride = args.swagger.params.hasOwnProperty('fileName') && args.swagger.params.fileName.value ? args.swagger.params.fileName.value : null;
+
+            if (document)
+            {
+                let documentMeta = await documentDAO.downloadDocumentGetMeta(constants.SECURE_ROLES, userName, document, fileNameOverride);
+
+                res.setHeader('Content-Length', documentMeta.size);
+                res.setHeader('Content-Type', documentMeta.metaData['content-type']);
+                res.setHeader('Content-Disposition', 'attachment;filename="' + documentMeta.fileName + '"');
+
+                return rp(documentMeta.url).pipe(res);
```

Comment:
What is 'rp' ?

Reply:
request-promise-native. I've updated the name and ensured the import is there